### PR TITLE
Verify if domain name has more than 50 chars

### PR DESCRIPTION
**Pull request by git-hyagi**
fixes: #4976

### CODE REVIEW

File: pulpcore/app/serializers/domain.py [hunk 1]
```diff
@@ -324,6 +324,11 @@ def validate_name(self, value):
         """Ensure name is not 'api' or 'content'."""
         if value.lower() in ("api", "content"):
             raise serializers.ValidationError(_("Name can not be 'api' or 'content'."))
+
+        """Ensure name has less than 50 characters."""
+        if len(value) > 50:
+            raise serializers.ValidationError(_("Name can have 50 characters at most."))
+
```
Comment (lubosmj) [inline pulpcore/app/serializers/domain.py:331]: I suppose we want something neat, like:
```
    name = serializers.SlugField(
        max_length="50",
        help_text=_("A name for this domain."),
        validators=[UniqueValidator(queryset=models.Domain.objects.all())],
    )
```

Comment (lubosmj) [reply pulpcore/app/serializers/domain.py:331]: A confusing part is that they claim `drf.SlugField` has `max_length` set to 50 by default: https://www.django-rest-framework.org/api-guide/fields/#slugfield.

However, there is no such a thing defined in the source code: https://github.com/encode/django-rest-framework/blob/41edb3b9dde2105b9fb015ce6b8171e7822baa30/rest_framework/fields.py#L796.

Therefore, I think setting `max_length=50` explicitly in the field will help.

Comment (mdellweg) [reply pulpcore/app/serializers/domain.py:331]: Good analysis. That sounds like a bug with drf (maybe docs only).

File: pulpcore/tests/functional/api/test_crud_domains.py [hunk 2]
```diff
@@ -268,6 +269,21 @@ def test_special_domain_creation(domains_api_client, gen_object_with_cleanup):
...
+    body = {
+        "name": name,
+        "storage_class": backend,
+        "storage_settings": storage_settings[backend],
+    }
+    try:
+        domain = gen_object_with_cleanup(domains_api_client, body)
+    except ApiException as e:
+        assert e.status == 400
+        assert "Ensure this field has no more than 50 characters." in e.body
```
Comment (lubosmj) [inline pulpcore/tests/functional/api/test_crud_domains.py:285]: Usually, we use `with pytest.raises(ApiException) as e`. But, this works for me too!

**Review by lubosmj — DISMISSED** [review]

File: pulpcore/tests/functional/api/test_crud_domains.py (hunk 2 above)
Comment (mdellweg) [reply pulpcore/tests/functional/api/test_crud_domains.py:285]: And we should do that.

**Review by lubosmj — APPROVED** [review]

**Review by mdellweg — APPROVED** [review]

**Pull request merged**